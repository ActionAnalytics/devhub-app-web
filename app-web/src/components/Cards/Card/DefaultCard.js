import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faExternalLinkSquareAlt } from '@fortawesome/free-solid-svg-icons';
import DotDotDot from 'react-dotdotdot';
import Image from 'react-image';
import styles from './Card.module.css';
import Card from '../../../hoc/Card';
import Link from '../../UI/Link/Link';
import { ARIA_LABEL_RESOURCE, ARIA_LABEL_REPO } from '../../../constants/ariaLabels';
import { CARD_CONFIG } from '../../../constants/ui';

const DefaultCard = ({
  sourcePath,
  sourceName,
  title,
  description,
  resourcePath,
  image,
  resourceType,
  author,
}) => (
  <Card resourceType={resourceType} title={title} author={author}>
    <div className={styles.Body}>
      <DotDotDot clamp={CARD_CONFIG.maxDescriptionLines} className={styles.BodyDescription}>
        <p>{description}</p>
      </DotDotDot>
      <Link to={resourcePath} aria-label={ARIA_LABEL_RESOURCE} className={styles.BodyImage}>
        <Image src={image} />
      </Link>
    </div>
  </Card>
);

DefaultCard.displayName = 'Default Card Component';

DefaultCard.propTypes = {
  sourcePath: PropTypes.string.isRequired,
  sourceName: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  resourcePath: PropTypes.string.isRequired,
  resourceType: PropTypes.string,
  image: PropTypes.string,
};

DefaultCard.defaultProps = {
  image: '',
  resourceType: '',
};

export default DefaultCard;
