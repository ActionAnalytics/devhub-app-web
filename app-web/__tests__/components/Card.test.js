import React from 'react';
import { shallow } from 'enzyme';
import DefaultCard from '../../src/components/Cards/Card/DefaultCard';

describe('Card Component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <DefaultCard title="title" description="description" resourcePath="resourcePath" />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
