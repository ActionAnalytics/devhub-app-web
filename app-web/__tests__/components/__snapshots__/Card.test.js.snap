// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Default Card Component
    description="description"
    image=""
    resourcePath="resourcePath"
    resourceType=""
    sourceName="sourceName"
    sourcePath="sourcePath"
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h2
          title="title"
        >
          <GatsbyLink
            aria-label="Learn more about this article"
            to="resourcePath"
          >
            <Dotdotdot
              clamp={2}
              tagName="span"
              truncationChar="…"
              useNativeClamp={true}
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f360",
                      "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                    ],
                    "iconName": "external-link-square-alt",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="1x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
              title
            </Dotdotdot>
          </GatsbyLink>
        </h2>,
        <div
          className="Body"
        >
          <Dotdotdot
            clamp={4}
            className="BodyDescription"
            tagName="div"
            truncationChar="…"
            useNativeClamp={true}
          >
            <p>
              description
            </p>
          </Dotdotdot>
          <GatsbyLink
            aria-label="Learn more about this article"
            className="BodyImage"
            to="resourcePath"
          >
            <Img
              container={[Function]}
              decode={true}
              loader={false}
              src=""
              unloader={false}
            />
          </GatsbyLink>
        </div>,
      ],
      "resourceType": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <GatsbyLink
            aria-label="Learn more about this article"
            to="resourcePath"
          >
            <Dotdotdot
              clamp={2}
              tagName="span"
              truncationChar="…"
              useNativeClamp={true}
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f360",
                      "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                    ],
                    "iconName": "external-link-square-alt",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="1x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
              title
            </Dotdotdot>
          </GatsbyLink>,
          "title": "title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "aria-label": "Learn more about this article",
            "children": <Dotdotdot
              clamp={2}
              tagName="span"
              truncationChar="…"
              useNativeClamp={true}
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f360",
                      "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                    ],
                    "iconName": "external-link-square-alt",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="1x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
              title
            </Dotdotdot>,
            "to": "resourcePath",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        448,
                        512,
                        Array [],
                        "f360",
                        "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                      ],
                      "iconName": "external-link-square-alt",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="1x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
                "title",
              ],
              "clamp": 2,
              "tagName": "span",
              "truncationChar": "…",
              "useNativeClamp": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f360",
                      "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                    ],
                    "iconName": "external-link-square-alt",
                    "prefix": "fas",
                  },
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": "1x",
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "title",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Dotdotdot
              clamp={4}
              className="BodyDescription"
              tagName="div"
              truncationChar="…"
              useNativeClamp={true}
            >
              <p>
                description
              </p>
            </Dotdotdot>,
            <GatsbyLink
              aria-label="Learn more about this article"
              className="BodyImage"
              to="resourcePath"
            >
              <Img
                container={[Function]}
                decode={true}
                loader={false}
                src=""
                unloader={false}
              />
            </GatsbyLink>,
          ],
          "className": "Body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <p>
                description
              </p>,
              "clamp": 4,
              "className": "BodyDescription",
              "tagName": "div",
              "truncationChar": "…",
              "useNativeClamp": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "description",
              },
              "ref": null,
              "rendered": "description",
              "type": "p",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "aria-label": "Learn more about this article",
              "children": <Img
                container={[Function]}
                decode={true}
                loader={false}
                src=""
                unloader={false}
              />,
              "className": "BodyImage",
              "to": "resourcePath",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "container": [Function],
                "decode": true,
                "loader": false,
                "src": "",
                "unloader": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h2
            title="title"
          >
            <GatsbyLink
              aria-label="Learn more about this article"
              to="resourcePath"
            >
              <Dotdotdot
                clamp={2}
                tagName="span"
                truncationChar="…"
                useNativeClamp={true}
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        448,
                        512,
                        Array [],
                        "f360",
                        "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                      ],
                      "iconName": "external-link-square-alt",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="1x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                title
              </Dotdotdot>
            </GatsbyLink>
          </h2>,
          <div
            className="Body"
          >
            <Dotdotdot
              clamp={4}
              className="BodyDescription"
              tagName="div"
              truncationChar="…"
              useNativeClamp={true}
            >
              <p>
                description
              </p>
            </Dotdotdot>
            <GatsbyLink
              aria-label="Learn more about this article"
              className="BodyImage"
              to="resourcePath"
            >
              <Img
                container={[Function]}
                decode={true}
                loader={false}
                src=""
                unloader={false}
              />
            </GatsbyLink>
          </div>,
        ],
        "resourceType": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <GatsbyLink
              aria-label="Learn more about this article"
              to="resourcePath"
            >
              <Dotdotdot
                clamp={2}
                tagName="span"
                truncationChar="…"
                useNativeClamp={true}
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        448,
                        512,
                        Array [],
                        "f360",
                        "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                      ],
                      "iconName": "external-link-square-alt",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="1x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                title
              </Dotdotdot>
            </GatsbyLink>,
            "title": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "aria-label": "Learn more about this article",
              "children": <Dotdotdot
                clamp={2}
                tagName="span"
                truncationChar="…"
                useNativeClamp={true}
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        448,
                        512,
                        Array [],
                        "f360",
                        "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                      ],
                      "iconName": "external-link-square-alt",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="1x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                title
              </Dotdotdot>,
              "to": "resourcePath",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          448,
                          512,
                          Array [],
                          "f360",
                          "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                        ],
                        "iconName": "external-link-square-alt",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size="1x"
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />,
                  "title",
                ],
                "clamp": 2,
                "tagName": "span",
                "truncationChar": "…",
                "useNativeClamp": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": Object {
                      "icon": Array [
                        448,
                        512,
                        Array [],
                        "f360",
                        "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                      ],
                      "iconName": "external-link-square-alt",
                      "prefix": "fas",
                    },
                    "inverse": false,
                    "listItem": false,
                    "mask": null,
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": "1x",
                    "spin": false,
                    "symbol": false,
                    "title": "",
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "title",
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Dotdotdot
                clamp={4}
                className="BodyDescription"
                tagName="div"
                truncationChar="…"
                useNativeClamp={true}
              >
                <p>
                  description
                </p>
              </Dotdotdot>,
              <GatsbyLink
                aria-label="Learn more about this article"
                className="BodyImage"
                to="resourcePath"
              >
                <Img
                  container={[Function]}
                  decode={true}
                  loader={false}
                  src=""
                  unloader={false}
                />
              </GatsbyLink>,
            ],
            "className": "Body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <p>
                  description
                </p>,
                "clamp": 4,
                "className": "BodyDescription",
                "tagName": "div",
                "truncationChar": "…",
                "useNativeClamp": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "description",
                },
                "ref": null,
                "rendered": "description",
                "type": "p",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "aria-label": "Learn more about this article",
                "children": <Img
                  container={[Function]}
                  decode={true}
                  loader={false}
                  src=""
                  unloader={false}
                />,
                "className": "BodyImage",
                "to": "resourcePath",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "container": [Function],
                  "decode": true,
                  "loader": false,
                  "src": "",
                  "unloader": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
